main/init/package.mill

```diff
@@ -86,4 +95,36 @@ object `package` extends RootModule with build.MillPublishScalaModule {
     )
     def testModuleDeps = super.testModuleDeps ++ Seq(build.scalalib, buildgen.test)
   }
+
+  object sbt extends build.MillPublishScalaModule {
+    def moduleDeps = Seq(buildgen, models(build.Deps.scalaVersion))
+
+    def sbtPluginJarResources = Task {
+      val assemblyPathRef = `sbt-mill-init-export-build`.assembly()
+      os.copy(assemblyPathRef.path, Task.dest / "sbt-mill-init-export-build-assembly.jar")
+      PathRef(Task.dest)
+    }
+    def resources: T[Seq[PathRef]] =
+      Task.Sources(super.resources() ++ Seq(sbtPluginJarResources()))
+    def testModuleDeps = super.testModuleDeps ++ Seq(build.scalalib, buildgen.test)
+
+    // An SBT plugin is built with Scala 2.12. See https://www.scala-sbt.org/1.x/docs/Plugins.html#Creating+an+auto+plugin.
+    object models
+        extends Cross[ModelsModule](build.Deps.sbtScalaVersion212, build.Deps.scalaVersion)
+    trait ModelsModule extends build.MillPublishCrossScalaModule {
+      // def moduleDeps = Seq(buildgen.tree())
+      def ivyDeps = Agg(build.Deps.upickle)
+      def compileIvyDeps = Agg(build.Deps.sbt) // for definition references only
+    }
+
+    // no need to publish this
+    // `test` fails with `build.MillScalaModule`, and since it's an SBT plugin project it's most likely not needed.
+    object `sbt-mill-init-export-build` extends ScalaModule {
```

Comment:
We can probably just call this `exportplugin`. It's already in `main.init.sbt` in the Mill codebase, so no need to repeat all of that again

Reply:
OK. I named it this way after sbt's [naming convention](https://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html#Artifact+naming+convention).

Shall I add a dash between the 2 words and make it `export-plugin`? Or what about still following the naming convetion and making it shorter like `sbt-export`?